sdk/v1/client_test.go

```diff
@@ -547,3 +547,109 @@ func TestClient_PatchJob(t *testing.T) {
 		})
 	})
 }
+
+func TestClient_GetTask(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	pathToCheck := "v1/jobs/883c81fd-726d-4ea3-9db8-7e7c781a01cc/tasks/zebedee"
+
+	headers := client.Headers{
+		IfMatch:          "*",
+		ServiceAuthToken: serviceToken,
+	}
+
+	Convey("Given clienter.Do doesn't return an error", t, func() {
+		body, err := json.Marshal(expectedTask)
+		if err != nil {
+			t.Errorf("failed to setup test data, error: %v", err)
+		}
+
+		header := http.Header{}
+		header.Add(ETagHeader, testETag)
+		httpClient := newMockHTTPClient(
+			&http.Response{
+				StatusCode: http.StatusCreated,
+				Body:       io.NopCloser(bytes.NewReader(body)),
+				Header:     header,
+			},
+			nil)
+
+		searchReindexClient := newSearchReindexClient(t, httpClient)
+
+		Convey("When search-reindexClient.GetTask is called", func() {
+			task, err := searchReindexClient.GetTask(ctx, headers, testJobID, testTaskName)
+			So(err, ShouldBeNil)
+
+			Convey("Then the expected jobid, task name, and number of documents, are returned", func() {
+				So(testJobID, ShouldEqual, expectedTask.JobID)
```

Comment:
I think this should test the returned job id:
```suggestion
				So(task.JobID, ShouldEqual, expectedTask.JobID)
```